tests/bluetooth/audio/mocks/src/fatal.c

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2023 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/fatal.h>
+#include <zephyr/kernel.h>
+
+void z_fatal_error(unsigned int reason, const z_arch_esf_t *esf)
+{
+	ztest_test_fail();
+}
```

Comment:
Is `kernel.h` needed?
Should we include `ztest.h` since we are using `ztest_test_fail`?

`ztest.h` may be missing from several files

Reply:
A little trick has been made. In CMakeLists.txt there is a line added:
```
target_compile_options(uut PRIVATE -std=c11 -include ztest.h)
```

`-include` is GCC preprocessor option to include the given header https://gcc.gnu.org/onlinedocs/gcc/Preprocessor-Options.html :) I copied it from other place (/tests/bluetooth/host/CMakeLists.txt)

Reply:
If this is a common approach in Zephyr, then that's fine :D It does somewhat go against the concept of "include what you use", but it's fine :)